src/React.re

```diff
@@ -300,46 +300,50 @@ module Event = {
   };
 };
 
-type element;
-type componentLike('props, 'return) = 'props => 'return;
-type component('props) = componentLike('props, element);
+type element('a); /* 'a captures any type information to differenciate with others 'a's */
+
+type keyed; /* Flag to mark element('a) to be keyed */
```

Comment:
just a note, I think this won't work correctly after https://github.com/ocaml/ocaml/pull/8900

Reply:
Totally, we can workaround it by adding "private" fields:

```
type keyed = private Tag_keyed
type non_keyed = private Tag_non_keyed
```